clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -1211,11 +1211,13 @@ llvm::Value *CodeGenFunction::emitCountedByPointerSize(
         getContext().getTypeSizeInChars(ElementTy->getPointeeType());
 
     if (ElementSize.isZero()) {
-      // This might be a __sized_by on a 'void *', which counts bytes, not
-      // elements.
+      // This might be a __sized_by (or __counted_by in GNU mode) on a
+      // 'void *', which counts bytes, not elements.
       auto *CAT = ElementTy->getAs<CountAttributedType>();
       if (!CAT || (CAT->getKind() != CountAttributedType::SizedBy &&
```

Comment:
Now, I think it can just be `if (!CAT) return std::optional<CharUnits>();` because now it's always `CharUnits::One()`. Maybe even just `assert(CAT)` if that's what it assumes here.